docs/Using-SRIOV-and-Clear-Containers.md

```diff
@@ -0,0 +1,218 @@
+01:00.1 Ethernet controller: Intel Corporation Ethernet Controller 10-Gigabit X540-AT2 (rev 01)
+```
+In our case, both 01:00.0 and 01:00.1 are the two ports on our x540-AT2 card 
+that we'll use.  You can use lshw to get further details on the controller 
+and verify it indeed supports SRIO-V.
+
+3.  Check how many VFs we can create:
+```
+$ sriov@sriov-1:~$ cat /sys/bus/pci/devices/0000\:01\:00.0/sriov_totalvfs
+63
+$ sriov@sriov-1:~$ cat /sys/bus/pci/devices/0000\:01\:00.1/sriov_totalvfs
+63
+```
+4.  Create VFs:
+
+Create virtual functions by editing sriov_numvfs. In our example, let's 
+just create one per physical device.  Note, this eliminates the usefulness 
+of SRIOV, and is just done for simplicity in this example so you needn't
+look at 128 virtual devices.
+
+```
+root@sriov-1:/home/sriov# echo 1 > /sys/bus/pci/devices/0000\:01\:00.0/sriov_numvfs
+root@sriov-1:/home/sriov# echo 1 > /sys/bus/pci/devices/0000\:01\:00.1/sriov_numvfs
+```
+
+5.  Verify that these were added to the host:
+
+```
+root@sriov-1:/home/sriov# lspci | grep Ethernet | grep Virtual
+02:10.0 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+02:10.1 Ethernet controller: Intel Corporation X540 Ethernet Controller Virtual Function (rev 01)
+```
+6.  Assign MAC address to each VF
+
+Depending on the NIC being used, you may need to explicitly set the MAC
+address for the Virtual Function device in order to guarantee that the 
+address is consistent on the host and when passed through to the guest.
+For example,
+
+```
+ip link set <pf> vf <vfidx> mac <my made up mac address>
+```
+
+## Example: Launch a Clear Container using SRIO-V
+
+1.  Build and start SRIOV plugin
+
+This assumes you already have GOPATH set in your environment.
+```
+sriov@sriov-2:~$ sudo mkdir /etc/docker/plugins
+sriov@sriov-2:~$ sudo cp go/src/github.com/clearcontainers/sriov/sriov.json /etc/docker/plugins/
+cd /go/src/github.com/clearcontainers/sriov
+go build
+sudo ./sriov &
+```
+2.  Create docker network
+
+To create the SRIOV based docker network, a subnet, vlanid and physical interface are required.
+
+```
+sudo docker network create -d sriov --internal --opt pf_iface=enp1s0f0 --opt vlanid=100 --subnet=192.168.0.0/24 vfnet
+
+
+E0505 09:35:40.550129    2541 plugin.go:297] Numvfs and Totalvfs are not same on the PF - Initialize numvfs to totalvfs
+ee2e5a594f9e4d3796eda972f3b46e52342aea04cbae8e5eac9b2dd6ff37b067
+```
+
+3.  Start containers and test connectivity
+
+To start a container making use of SRIOV:
+
+```
+sriov@sriov-2:~$ sudo docker run --runtime=cc-runtime --net=vfnet --ip=192.168.0.10 -it alpine
+```
+
+If you have two machines with SRIOV enabled NICs connected back to back and each
+has a network with matching vlan-id created, you can test connectivity as follows:
+
+Machine #1:
+```
+sriov@sriov-2:~$ sudo docker run --runtime=cc-runtime --net=vfnet --ip=192.168.0.10 -it mcastelino/iperf iperf3 -s
```

Comment:
We need to explain why the `mcastelino/iperf` docker image is being used here. Is it required? If so, why?

Reply:
@mcastelino what do you think about this?
You mentioned today about having a Docker image with a Dockerfile with all the necessary tools.